### PR TITLE
For #20872: Prevent crash when updating pending bookmarks to be deleted without a selected folder.

### LINKED ISSUE #20872: NullPointerException BookmarkFragment in updatePendingBookmarksToDelete
Low volume.
 
Sentry :
* Nightly https://sentry.prod.mozaws.net/operations/firefox-nightly/issues/12052102/
* Beta https://sentry.prod.mozaws.net/operations/firefox-beta/issues/11800404
* Release https://sentry.prod.mozaws.net/operations/firefox/?query=is%3Aunresolved+updatePendingBookmarksToDelete

```
java.lang.NullPointerException: null
    at org.mozilla.fenix.library.bookmarks.BookmarkFragment.updatePendingBookmarksToDelete(BookmarkFragment.kt:2)
    at org.mozilla.fenix.library.bookmarks.BookmarkFragment.deleteMulti(BookmarkFragment.kt:5)
    at org.mozilla.fenix.library.bookmarks.BookmarkFragment.access$deleteMulti(BookmarkFragment.kt:1)
    at org.mozilla.fenix.library.bookmarks.BookmarkFragment$onCreateView$4.invoke(BookmarkFragment.kt:3)
    at org.mozilla.fenix.library.bookmarks.DefaultBookmarkController.handleBookmarkDeletion(BookmarkController.kt:1)
    at org.mozilla.fenix.library.bookmarks.BookmarkFragmentInteractor.onDelete(BookmarkFragmentInteractor.kt:15)
    at org.mozilla.fenix.library.bookmarks.viewholders.BookmarkNodeViewHolder$1.invoke(BookmarkNodeViewHolder.kt:5)
    at org.mozilla.fenix.library.bookmarks.BookmarkItemMenu$menuItems$6.invoke(BookmarkItemMenu.kt:1)
    at mozilla.components.browser.menu2.adapter.TextMenuCandidateViewHolder.onClick(TextMenuCandidateViewHolder.kt:1)
    at android.view.View.performClick(View.java:7217)
    at android.view.View.performClickInternal(View.java:7191)
    at android.view.View.access$3500(View.java:828)
    at android.view.View$PerformClick.run(View.java:27679)
    at android.os.Handler.handleCallback(Handler.java:900)
    at android.os.Handler.dispatchMessage(Handler.java:103)
    at android.os.Looper.loop(Looper.java:219)
    at android.app.ActivityThread.main(ActivityThread.java:8349)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)
```

### PR DESCRIPTION
**Pull request by Amejia481**
Fixes #20872 
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt [hunk 1]
```diff
@@ -409,7 +409,8 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), UserInteractionHan
 
     private fun updatePendingBookmarksToDelete(selected: Set<BookmarkNode>) {
         pendingBookmarksToDelete.addAll(selected)
-        val bookmarkTree = sharedViewModel.selectedFolder!! - pendingBookmarksToDelete
+        val selectedFolder = sharedViewModel.selectedFolder ?: return
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt:412]: do we need to log the NPE, or is this info we don't really care about?

**Review by MozillaNoah — APPROVED** [review]

**Comment by gabrielluong** [comment]
Let's make sure we add a qa-needed when this lands

**Comment by Amejia481** [comment]
👍🏽

File: app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt:412]: Thanks for bringing that up I think we should investigate if we should refactor how we deal with bookmarks, I noticed there are other usages of `selectedFolder!!` that could cause NPE, I don't have the whole context around why it was built in that way, for now I think we should stop the crash until we do the refactor,  I filed https://github.com/mozilla-mobile/fenix/issues/24896 for investigating.

**Comment by gabrielluong** [comment]
Removing needs-landing since it's blocking the queue

**Comment by Amejia481** [comment]
It looks it was a network issue, trying to re-trigger.
```
[task 2022-04-20T21:46:39.604Z] * What went wrong:
[task 2022-04-20T21:46:39.604Z] Execution failed for task ':app:getDependencies'.
[task 2022-04-20T21:46:39.604Z] > Failed to notify dependency resolution listener.
[task 2022-04-20T21:46:39.604Z]    > Problems reading data from Binary store in /builds/worker/.gradle/.tmp/gradle1386446114597824923.bin offset 1570 exists? true
[task 2022-04-20T21:46:39.604Z]    > Could not resolve all dependencies for configuration ':app:debugApiDependenciesMetadata'.
[task 2022-04-20T21:46:39.604Z]       > Problems reading data from Binary store in /builds/worker/.gradle/.tmp/gradle1386446114597824923.bin offset 1570 exists? true
[task 2022-04-20T21:46:39.604Z] 
[task 2022-04-20T21:46:39.604Z] * Try:
[task 2022-04-20T21:46:39.604Z] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[task 2022-04-20T21:46:39.605Z] 
[task 2022-04-20T21:46:39.605Z] * Get more help at https://help.gradle.org

```

**Pull request merged**